packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -2695,32 +2683,43 @@ export class ModularEditBuilder extends EditBuilder<ModularChangeset> {
 
 		this.applyChange(
 			tagChange(
-				buildModularChangesetFromNode(
+				buildModularChangesetFromNode({
 					path,
 					nodeChange,
-					newTupleBTree(),
-					newTupleBTree(),
-					newCrossFieldKeyTable(),
-					this.idAllocator,
+					nodeChanges: newTupleBTree(),
+					nodeToParent: newTupleBTree(),
+					crossFieldKeys: newCrossFieldKeyTable(),
+					idAllocator: this.idAllocator,
 					revision,
-				),
+				}),
 				revision,
 			),
 		);
 	}
 }
 
-function buildModularChangesetFromField(
-	path: FieldUpPath,
-	fieldChange: FieldChange,
-	nodeChanges: ChangeAtomIdBTree<NodeChangeset>,
-	nodeToParent: ChangeAtomIdBTree<FieldId>,
-	crossFieldKeys: CrossFieldKeyTable,
-	idAllocator: IdAllocator = idAllocatorFromMaxId(),
-	localCrossFieldKeys: CrossFieldKeyRange[] = [],
-	childId: NodeId | undefined = undefined,
-	revision: RevisionTag,
-): ModularChangeset {
+function buildModularChangesetFromField(props: {
+	path: FieldUpPath;
+	fieldChange: FieldChange;
+	nodeChanges: ChangeAtomIdBTree<NodeChangeset>;
+	nodeToParent: ChangeAtomIdBTree<FieldId>;
+	crossFieldKeys: CrossFieldKeyTable;
+	localCrossFieldKeys?: CrossFieldKeyRange[];
+	revision: RevisionTag;
+	idAllocator?: IdAllocator;
+	childId?: NodeId;
+}): ModularChangeset {
+	const {
+		path,
+		fieldChange,
+		nodeChanges,
+		nodeToParent,
+		crossFieldKeys,
+		idAllocator = idAllocatorFromMaxId(),
```

Comment:
I've never seen this kind of notation before. Does this ensure that `idAllocator` gets the value from `props` if one is provided?
Also, do you know if `idAllocatorFromMaxId()` gets run if `props.idAllocator` is populated?

Reply:
Yes, if it's provided, it will use the provided value instead. And if `props.idAllocator` is passed, `idAllocatorFromMaxid()` will not be called.
I verified this via TS playground. We also have this pattern in our codebase. See example - https://github.com/microsoft/FluidFramework/blob/7b7e736b8c6fa5b426ca813c39f3f2fc5f3c9a9c/packages/runtime/container-runtime/src/containerRuntime.ts#L3511

Reply:
Thanks really cool, thanks for showing me this.